doc/nyan.md

```diff
@@ -344,23 +344,22 @@ the calculation is done like this:
 
 * Members of `nyan::Object` must have a type, which can be a
   * primitive type
-    - `text`:     `"lol"`          - (duh.)
-    - `int`:      `1337`           - (some number)
-    - `float`:    `42.235`, `inf`  - (some floating point number)
-    - `bool`:     `true`, `false`  - (some boolean value)
-    - `file`:     `"./name" `      - (some filename, relative to the directory
-                                      the defining nyan file is located at.
-                                      If absolute, the path is relative to
-                                      an engine defined root directory)
+    - `text`:     `"lol"`                  - (duh.)
+    - `int`:      `1337`                   - (some number)
+    - `float`:    `42.235`, `inf`, `-inf`  - (some floating point number)
+    - `bool`:     `true`, `false`          - (some boolean value)
+    - `file`:     `"./name" `              - (some filename, relative to the directory the defining nyan file is located at.
+                                              If absolute, the path is relative to an engine defined root directory.)
   * ordered set of elements of a type: `orderedset(type)`
   * set of elements of a type: `set(type)`
+  * dictionary of elements of a type to elements of another type: `dict(keyType, valueType)`
```

Comment:
i was always unsure about adding a dict-type, because it may have &lt;gmanvoice&gt; *unforseen consequences* &lt;/gmanvoice&gt;:
Currently you can implement dicts by using a set of objects, and the objects then have some values. Those objects are always referencable, as they have a name. Thus they can always be patched. ~~When we add a real key-value type, values can get lost.~~ If `valueType` is an object, the object is always referencable. Non-object values can get lost when it is updated in the dict. Which may not be desirable.

I guess that we should add dicts, though. Intuitively they seem a good addition, and loosing non-object values is probably okay, because who cares for a `"random string"` or a number like `235`. If you overwrite it, you likely won't revert.

Reply:
Yes but its huge

``` python
Something():
    FoodCost(ResourceAmount):
        resource: Food
        amount: 50
    WoodCost(ResourceAmount):
        resource: Wood
        amount: 50
    cost: {FoodCost, WoodCost}
```

My first thought was some kind of an anonymous object, but then you have something that cannot be referenced so why not just dicts.

Also having a bunch of prebuild WoodCost50, WoodCost75 but again you cannot patch them.

Maybe having a way to inline it, but then it gets ugly...